verifier/plonk/gates/exponentiation.go

```diff
@@ -0,0 +1,105 @@
+package gates
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Electron-Labs/plonky2-groth16-verifier/goldilocks"
+	"github.com/consensys/gnark/frontend"
+)
+
+type ExponentiationGate struct {
+	NumPowerBits int `json:"num_power_bits"`
+}
+
+func NewExponentiationGate(id string) *ExponentiationGate {
+	id = strings.Split(id, ", _phantom")[0]
```

Comment:
I think we should have an equality check here that the split string is actually `": PhantomData<plonky2_field::goldilocks_field::GoldilocksField> }"`